webpack/InsightsCloudSync/Components/InsightsTable/InsightsTableActions.js

```diff
@@ -51,15 +54,24 @@ export const fetchInsights = (queryParams = {}) => (dispatch, getState) => {
     })
   );
 
-  const isAllSelected = selectIsAllSelected(state);
-  if (isAllSelected) {
+  if (!isSelectAll) {
     dispatch(clearAllSelection());
-  } else {
-    dispatch(setSelectAllAlert(false));
-    dispatch(setSelectAll(false));
   }
 };
 
+const selectAllIdsFromResonse = (dispatch, response) => {
+  const selectedIds = {};
+  const results = response?.data?.hits || [];
+
+  dispatch(selectAll());
+  results.forEach(row => {
+    selectedIds[row.id] = true;
+  });
+  dispatch(selectByIds(selectedIds));
+  dispatch(setSelectAllAlert(true));
+  dispatch(selectAll());
```

Comment:
Why do you need to dispatch `selectAll` twice?

Reply:
I don't, removed